composer/datasets/ade20k.py

```diff
@@ -112,6 +140,87 @@ def build_ade20k_dataloader(
     )
 
 
+def build_streaming_ade20k_dataloader(
+    batch_size: int,
+    remote: str,
+    *,
+    version: int = 2,
+    local: str = '/tmp/mds-cache/mds-ade20k/',
+    split: str = 'train',
+    drop_last: bool = True,
+    shuffle: bool = True,
+    base_size=512,
+    min_resize_scale: float = 0.5,
+    max_resize_scale: float = 2.0,
+    final_size: int = 512,
+    ignore_background: bool = True,
+    **dataloader_kwargs,
+):
+    """Build an ADE20k streaming dataset.
+
+    Args:
+        batch_size (int): Batch size per device.
+        remote (str): Remote directory (S3 or local filesystem) where dataset is stored.
+        version (int): Which version of streaming to use. Default: ``1``.
+        local (str): Local filesystem directory where dataset is cached during operation.
+            Default: ``'/tmp/mds-cache/mds-ade20k/```.
+        split (str): The dataset split to use, either 'train' or 'val'. Default: ``'train```.
+        base_size (int): Initial size of the image and target before other augmentations. Default: ``512``.
+        min_resize_scale (float): The minimum value the samples can be rescaled. Default: ``0.5``.
+        max_resize_scale (float): The maximum value the samples can be rescaled. Default: ``2.0``.
+        final_size (int): The final size of the image and target. Default: ``512``.
+        ignore_background (bool): If true, ignore the background class when calculating the training loss.
+            Default: ``true``.
+        **dataloader_kwargs (Dict[str, Any]): Additional settings for the dataloader (e.g. num_workers, etc.)
+    """
+    if version == 1:
+        warn_streaming_dataset_deprecation(old_version=version, new_version=2)
+        dataset = StreamingADE20k(remote=remote,
+                                  local=local,
+                                  split=split,
+                                  shuffle=shuffle,
+                                  base_size=base_size,
+                                  min_resize_scale=min_resize_scale,
+                                  max_resize_scale=max_resize_scale,
+                                  final_size=final_size,
+                                  batch_size=batch_size)
+    elif version == 2:
+
+        try:
+            import streaming
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='streaming', conda_package='mosaicml-streaming') from e
+
+        # Build the sets of transformations for ADE20k
+        all_transforms = build_ade20k_transformations(split=split,
+                                                      base_size=base_size,
+                                                      min_resize_scale=min_resize_scale,
+                                                      max_resize_scale=max_resize_scale,
+                                                      final_size=final_size)
+
+        dataset = streaming.vision.ADE20K(remote=remote,
```

Comment:
I would recommend to pass the `batch_size` to streaming ADE20K class. When `batch_size` is provided, worker indices will be constructed so that there is at most one incomplete batch at the end of each epoch.